multihash/src/errors.rs

```diff
@@ -1,24 +1,75 @@
 use std::{fmt, error};
 
-#[derive(Debug)]
-pub enum Error {
+/// Error that can happen when encoding some bytes into a multihash.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum EncodeError {
+    /// The requested hash algorithm isn't supported by this library.
     UnsupportedType,
+}
+
+impl fmt::Display for EncodeError {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str(error::Error::description(self))
+    }
+}
+
+impl error::Error for EncodeError {
+    #[inline]
+    fn description(&self) -> &str {
```

Comment:
I think `description` is ["soft-deprecated"](https://doc.rust-lang.org/std/error/trait.Error.html#method.description), so the `std::error::Error` impl could be simplified further.